core/runtime/jsruntime.rs

```diff
@@ -1304,12 +1304,28 @@ impl JsRuntime {
   pub async fn call_and_await(
     &mut self,
     function: &v8::Global<v8::Function>,
+  ) -> Result<v8::Global<v8::Value>, Error> {
+    self.call_with_args_and_await(function, &[]).await
+  }
+
+  /// Call a function with args. If it returns a promise, run the event loop until that
+  /// promise is settled. If the promise rejects or there is an uncaught error
+  /// in the event loop, return `Err(error)`. Or return `Ok(<await returned>)`.
+  pub async fn call_with_args_and_await(
+    &mut self,
+    function: &v8::Global<v8::Function>,
+    args: &[v8::Global<v8::Value>],
   ) -> Result<v8::Global<v8::Value>, Error> {
     let promise = {
       let scope = &mut self.handle_scope();
+      let mut local_args: SmallVec<[v8::Local<v8::Value>; 32]> =
+        SmallVec::with_capacity(args.len());
+      for v in args {
+        local_args.push(v8::Local::new(scope, v));
+      }
```

Comment:
Please let me know if there is a better way to do this! I'm new to Rust and I copied this from similar code in the same file (choosing `32` arbitrarily), so not sure if this looks right.